gadget.yaml

```diff
@@ -14,15 +14,29 @@ volumes:
         offset-write: mbr+92
         content:
           - image: pc-core.img
-      - name: EFI System
+      - name: Recovery
+        role: system-recovery
+        filesystem: vfat
+        # UEFI will boot the ESP partition by default first
         type: EF,C12A7328-F81F-11D2-BA4B-00A0C93EC93B
+        filesystem-label: ubuntu-seed
+        size: 1200M
+        content:
+          - source: grubx64.efi
+            target: EFI/boot/grubx64.efi
+          - source: shim.efi.signed
+            target: EFI/boot/bootx64.efi
+          - source: grub.cfg-recovery
+            target: EFI/ubuntu/grub.cfg
+      - name: System boot
+        type: 0B,EBD0A0A2-B9E5-4433-87C0-68B6B72699C7
         filesystem: vfat
-        filesystem-label: system-boot
-        size: 50M
+        filesystem-label: ubuntu-boot
+        size: 100M
```

Comment:
I'm skeptical of this as a size for the partition given that we will need to store 2x vmlinuz (+ embedded initramfs) on this fs for the FDE case.

Reply:
What size do you have in mind? I just checked and the kernel is right now around 8Mb for vmlinuz and around 7Mb for the initamfs (this includes some basic tools to unlock the LUKS device).

Reply:
initramfs will include all of the microcode updates, which are currently ~2.5M; one golang binary (snapd is currently 17M on amd64); systemd (~2M or so?); C runtime (at least 14M, based on the library dependencies of systemd+udev).

The microcode doesn't get compressed.  The other bits may get compressed, but there's an ongoing discussion whether compressing them negatively impacts boot speed, so we shouldn't assume they are compressed (or compressed much).

That's 35.5M for the initramfs, not counting any kernel modules which might need to be shipped in the embedded initramfs instead of built in to the kernel image; plus 8.4M for vmlinuz, which puts us at ~44M.

If we expect these systems to be usable for 10 years, we'll need to expect some expansion of kernels, if only due to the addition of drivers for new hardware.

So 150 or 200M seems more suitable to me (rounded to 4MiB).

Reply:
30M	ubuntu-core-initramfs.img
9.0M /boot/vmlinuz-5.2.0-8-generic
is what i have currently, without firmware, microcode, snapd. But this has too many kernel modules in the initrd.